Dependencies/AndroidExtensions/Source/JavaWrappers.cpp

```diff
@@ -108,12 +108,16 @@ namespace java::lang
 
     String::operator std::string() const
     {
-        return m_env->GetStringUTFChars(m_string, nullptr);
+        const char* buffer{m_env->GetStringUTFChars(m_string, nullptr)};
```

Comment:
the `nullptr` is for isCopy right? Didn't we say we need to get this result to know if we need to release or not?

Reply:
Rereading the docs, I don't think so. They say this:

> Returns a pointer to an array of bytes representing the string in modified UTF-8 encoding. **This array is valid until it is released by ReleaseStringUTFChars().**

I think this means you must release the buffer regardless, and this StackOverflow suggests the same: https://stackoverflow.com/questions/5859673/should-you-call-releasestringutfchars-if-getstringutfchars-returned-a-copy

I think the JVM docs also agree with this, though they are confusing. For example, on one page it says:

> You must call Release<Type> once for every Get<Type> call, regardless of the value of the isCopy parameter. This step is necessary because calling Release<Type> deletes JNI local references that might otherwise prevent garbage collection.

This seems pretty clear. But then on another page it says:

> Ensure that you free copies made by a Get<Type> function by calling the corresponding Release<Type> function when the copy is no longer needed.

Which could suggest that a Release* call is exclusively needed for copies, or could just mean that a Release* call is inclusively needed for copies.

https://www.ibm.com/support/knowledgecenter/SSYKE2_7.0.0/com.ibm.java.win.70.doc/diag/understanding/jni_mode_flag.html

Reply:
hmm, okay.